src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/ScanCommandCallable.java

```diff
@@ -58,6 +60,13 @@ public ScanCommandOutput call() {
             cmd.add(scanCommand.getTargetPath());
 
             commandToExecute = createPrintableCommand(cmd);
+            
+            if (OperatingSystemType.determineFromSystem().equals(OperatingSystemType.WINDOWS) 
+            		&& commandToExecute.length() > WINDOWS_CHARACTER_LIMIT) {
```

Comment:
Here we are checking the length of commandToExecute yet it seems to me like it is not the actual command that ends up executed (its content is generated using `createPrintableCommand()` which changes it somewhat.

Based on the stack trace in the ticket a safer way to do this may be to try to execute the command and catch `java.io.IOException: CreateProcess error=206`. What do you think?

Reply:
Yes, cmd is the actual command that is run but it was a bit trickier to calculate the length of that than the also available commandToExecute. That one is changed as you say but I believe all that createPrintableCommand does is change sensitive password fields. For example, if the supplied password was xyz we would substitute *** for it...my reading of the code was that we would have the same number of characters either way but there is something to be said about calculating on cmd as that is the actual command and perhaps things would deviate in the future.

This raises the point the limit I'm using is a combination of research and observed behavior. I didn't find a nice boundary like exactly 32k characters. 32,764 often comes up when googling but I found I had to lower that a bit as I kept getting failures.

My main approach had been to go after the concept of check the command length and prevent approach mentioned in the Jira issue. I think that prevents a bit of wasteful invocation time but I'm kind of coming around to your idea as it seems a bit safer and cleaner. Thanks for raising this and starting this conversation! @gopannabd do you have thoughts here?

Reply:
It is a good idea to capture the exception to ensure we have coverage across OSs.

Reply:
@gopannabd @andrian-synopsys could you please take another look? I switched to the catch approach. Some relevant information: a) this error appears to be Windows only but I didn't see much advantage to putting a windows check around this and b) I could have caught the IOException in its own catch but this made the error message construction a tiny bit easier but I could go either way. Thoughts?